pkg/k8smgmt/cluster.go

```diff
@@ -394,6 +413,72 @@ func convertNodeResource(res v1.ResourceName, quantity resource.Quantity) (strin
 	return name, dec, nil
 }
 
+func getNodeGPUInfo(labels map[string]string) ([]*edgeproto.GPUResource, *edgeproto.GPUSoftwareInfo, error) {
+	// Note: we only support one GPU per node, as not sure
+	// what the labels would look like for multiple GPUs
+	// or if the gpu-operator even supports it.
+	gpu := &edgeproto.GPUResource{}
+	gpuFound := false
+	sw := &edgeproto.GPUSoftwareInfo{}
+	for k, v := range labels {
+		switch k {
+		case "nvidia.com/gpu.product":
+			gpu.ModelId = ensureGPUPrefix(v, cloudcommon.GPUVendorNVIDIA)
+			gpu.Vendor = cloudcommon.GPUVendorNVIDIA
+			gpuFound = true
+		case "nvidia.com/gpu.memory":
+			val, err := strconv.ParseUint(v, 10, 64)
+			if err != nil {
+				return nil, nil, fmt.Errorf("failed to parse gpu memory, %s, %v", v, err)
+			}
+			// nvidia memory is specified in MB
+			gpu.Memory = val / 1024
+		case "nvidia.com/gpu.replicas":
```

Comment:
`gpu.replicas` might be misleading, as you can change the replicas and that would allow for time-sharing of the gpu. Do we want to support this? The metrics for the gpu and the attribution would be skewed. I definitely think we should track replicas to support time slicing, but if we want to assign whole gpus, `gpu.count` might work better.

Reply:
You're right, I believe I should be using gpu.count instead. I'll fix that.